model/DataBaseAccess.php

```diff
@@ -362,19 +363,25 @@ public function removeAllPermissions($resourceIds)
      */
     public function checkPermissions($userIds)
     {
-        $inQueryUser = implode(',', array_fill(0, count($userIds), ' ? '));
-        $query = sprintf(
-            'SELECT %s FROM %s WHERE %s IN (%s)',
-            self::COLUMN_USER_ID,
-            self::TABLE_PRIVILEGES_NAME,
-            self::COLUMN_USER_ID,
-            $inQueryUser
-        );
-        $results = $this->fetchQuery($query, array_values($userIds));
-        foreach ($results as $result) {
-            $existsUsers[$result[self::COLUMN_USER_ID]] = $result[self::COLUMN_USER_ID];
+        $chunks = array_chunk($userIds, $this->getOption(self::OPTION_FETCH_USER_PERMISSIONS_CHUNK_SIZE, 10));
```

Comment:
suggestion: I think it may be safe to use a higher default for this: We are already inserting 20k rows per query at [DataBaseAccess.php](https://github.com/oat-sa/extension-tao-dac-simple/blob/develop/model/DataBaseAccess.php#L449) (with 3 params per insert, so 20k * 3 ~= 60k parameters per query, which is less than the [max of 64k params](https://klotzandrew.com/blog/postgres-passing-65535-parameter-limit) / [32k params](https://stackoverflow.com/questions/54226729/is-there-any-possibility-for-a-workaround-to-postgresql-32760-bind-parameters-li)  typically found in Postgres and other RDBMS.

Probably 100 or even 1000 is still pretty safe.

Reply:
It fails with 50 now, it was really strange for me too, but it only works with lower limit. Keep in mind that here we are not inserting, but getting permission to a set of users (UDIR will not have more than 90) per resource, so this list should be maximum 90. It is also configurable, so we can increase it in the future if needed. I am afraid if we increase it to more than 10 we will can have a big response input/output that can cause more issues.

Reply:
The problem seems to be the result set size, without a limit, we were having millions of rows as result.